airbyte-integrations/connectors/source-asana/source_asana/streams.py

```diff
@@ -1,423 +0,0 @@
-    argument in request.
-    """
-
-    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
-        yield from self.read_slices_from_records(stream_class=Projects, slice_field="project_gid")
-
-
-class AttachmentsCompact(AsanaStream):
-    use_cache = True
-
-    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
-        return "attachments"
-
-    def request_params(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
-        params = super().request_params(**kwargs)
-        params["parent"] = stream_slice["parent_gid"]
-        return params
-
-    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
-        yield from self.read_slices_from_records(stream_class=Projects, slice_field="parent_gid")
-        yield from self.read_slices_from_records(stream_class=Tasks, slice_field="parent_gid")
-
-
-class Attachments(AsanaStream):
-    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
-        attachment_gid = stream_slice["attachment_gid"]
-        return f"attachments/{attachment_gid}"
-
-    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
-        yield from self.read_slices_from_records(stream_class=AttachmentsCompact, slice_field="attachment_gid")
-
-    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        response_json = response.json()
-        section_data = response_json.get("data", {})
-        if isinstance(section_data, dict):  # Check if section_data is a dictionary
-            yield section_data
-        elif isinstance(section_data, list):  # Check if section_data is a list
-            yield from section_data
-
-
-class CustomFields(WorkspaceRelatedStream):
-    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
-        workspace_gid = stream_slice["workspace_gid"]
-        return f"workspaces/{workspace_gid}/custom_fields"
-
-
-class Events(AsanaStream):
-    primary_key = "created_at"
-    sync_token = None
-
-    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
-        return "events"
-
-    def read_records(self, *args, **kwargs):
-        # Check if sync token is available
-        if self.sync_token is not None:
-            # Pass the sync token as a request parameter
-            kwargs["next_page_token"] = {"sync": self.sync_token}
-
-        yield from super().read_records(*args, **kwargs)
-
-        # After reading records, update the sync token
-        self.sync_token = self.get_latest_sync_token()
-
-    def get_latest_sync_token(self) -> str:
-        latest_sync_token = self.state.get("last_sync_token")  # Get the previous sync token
-
-        if latest_sync_token is None:
-            return None
-
-        return latest_sync_token["sync"]  # Extract the sync token value
```

Comment:
this is a breaking change since the format of the state messages is going to change. Let's flag this as a breaking change specific to the Events stream